From 0a4a16d11cc263c3f32325f985b9ed94b04af79f Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Wed, 22 Feb 2017 19:25:13 -0500
Subject: [PATCH] allow page size to vary on arm

the ABI for arm was silently changed at some point to allow page sizes
other than 4k; traditional binaries built with only 4k-aligned offsets
between load segments cannot run on such systems, but newer binutils
versions use 64k offset alignment.

while larger page size is undesirable for various reasons, users have
encountered hardware and/or kernels that lock the page size to a
larger value, so follow the new ABI and allow it to vary.
---
 arch/arm/bits/limits.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/bits/limits.h b/arch/arm/bits/limits.h
index 65a3dd64..fbc6d238 100644
--- a/arch/arm/bits/limits.h
+++ b/arch/arm/bits/limits.h
@@ -1,6 +1,5 @@
 #if defined(_POSIX_SOURCE) || defined(_POSIX_C_SOURCE) \
  || defined(_XOPEN_SOURCE) || defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
-#define PAGE_SIZE 4096
 #define LONG_BIT 32
 #endif
 
-- 
2.11.1

